web/src/app/decks/[deck_id]/page.tsx

```diff
@@ -106,6 +144,127 @@ export default function DeckPage({
         }
     }
 
+    const updateDeckMutation = useMutation({
+        mutationFn: (values: z.infer<typeof deckUpdateSchema>) =>
+            updateDeckDecksDeckIdPatch({
+                path: { deck_id: deck_id },
+                body: values,
+            }),
+        onSuccess: () => {
+            queryClient.invalidateQueries({ queryKey: ['decks', deck_id] });
+            setIsEditing(false);
+            // TODO: Add toast notification
+        },
+        onError: (error) => {
+            // TODO: Add error handling/toast
+            console.error('Failed to update deck:', error);
+        },
+    });
+
+    const togglePauseDeckMutation = useMutation({
+        mutationFn: (deck: DeckOut) =>
+            updateDeckDecksDeckIdPatch({
+                path: { deck_id: deck.id },
+                body: { is_paused: !deck.is_paused },
+            }),
+        onSuccess: () => {
+            queryClient.invalidateQueries({ queryKey: ['decks', deck_id] });
+        },
+    });
+
+    const exportDeckMutation = useMutation({
+        mutationFn: async (deck: DeckOut) => {
+            // Use fetch client directly due to issue with hey-api
+            // https://github.com/hey-api/openapi-ts/issues/804
+            const response = await fetch(`${apiUrl}/decks/${deck.id}/export`, {
+                method: 'GET',
+                credentials: 'include',
+            });
+
+            if (!response.ok) {
+                throw new Error(`Export failed: ${response.status}`);
+            }
+
+            const disposition = response.headers.get('Content-Disposition');
+            const filenameMatch = disposition?.match(/filename="(.+)"/);
+            const filename = filenameMatch?.[1] ?? 'deck.json';
+
+            const blob = await response.blob();
+            return { blob, filename };
+        },
+        onSuccess: ({ blob, filename }) => {
+            const url = window.URL.createObjectURL(blob);
+            const link = document.createElement('a');
+            link.href = url;
+            link.download = filename;
+            document.body.appendChild(link);
+            link.click();
+
+            document.body.removeChild(link);
+            window.URL.revokeObjectURL(url);
+        },
+    });
```

Comment:
Is this a mutation or "just" a download?

Reply:
using react-query to keep it consistent